backend/src/util/virtualLinkTransaction.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Transaction } from '@model/Transaction'
+import { SaveOptions, RemoveOptions } from '@dbTools/typeorm'
+import { Transaction as dbTransaction } from '@entity/Transaction'
+import { TransactionTypeId } from '@enum/TransactionTypeId'
+import { User } from '@model/User'
+import Decimal from 'decimal.js-light'
+
+const virtualLinkTransaction = (
+  balance: Decimal,
+  amount: Decimal,
+  holdAvailableAmount: Decimal,
+  decay: Decimal,
+  createdAt: Date,
+  validUntil: Date,
+  user: User,
+): Transaction => {
+  const linkDbTransaction: dbTransaction = {
+    id: -2,
+    userId: -1,
+    previous: -1,
+    typeId: TransactionTypeId.TRANSACTION_LINK,
+    amount: amount,
+    balance: balance.minus(holdAvailableAmount.toString()),
+    balanceDate: validUntil,
+    decayStart: createdAt,
```

Comment:
will those reflect proper min and max values or any valuable information or should we null those?

Reply:
see the better DB query